ddl/ddl_tiflash_api.go

```diff
@@ -463,6 +511,125 @@ func (d *ddl) pollTiFlashReplicaStatus(ctx sessionctx.Context, pollTiFlashContex
 	return allReplicaReady, nil
 }
 
+func (d *ddl) pollTiFlashPeerInfo(ctx sessionctx.Context, pollTiFlashContext *TiFlashManagementContext) (bool, error) {
+	allReplicaReady := true
+
+	if len(pollTiFlashContext.TiFlashStores) == 0 {
+		return true, nil
+	}
+
+	// Start to process every table.
+	schema := d.GetInfoSchemaWithInterceptor(ctx)
+	if schema == nil {
+		return false, errors.New("Schema is nil")
+	}
+
+	var tableMap = make(map[int64]TiFlashReplicaStatus)
+	// Collect TiFlash Replica info, for every table.
+	for _, db := range schema.AllSchemas() {
+		tbls := schema.SchemaTables(db.Name)
+		for _, tbl := range tbls {
+			tblInfo := tbl.Meta()
+			GetTiFlashReplicaMapInfo(tblInfo, &tableMap)
+		}
+	}
+
+	tiflashStoreIDs := make([]int64, len(pollTiFlashContext.TiFlashStores))
```

Comment:
How about using a map?

Reply:
it used to grep the invaildTableIDs..

Reply:
Maybe you can also build a map for `invalidPeerIDs`, and then check whether the two map are equal (length is identical && everything in map0 is in map1).